File: src/dataset/cpp/prime-factors_exercism-toemm/gemini/GeminiFlash_prime_factors_v4.cpp
#include "prime_factors.h"

#include <vector>
#include <cmath>

namespace prime_factors {

using namespace std;


vector<long long> of(long long n) {

    vector<long long> factors;

    if (n <= 1) {
        return factors;
    }

    while (n % 2 == 0) {
        factors.push_back(2);
        n /= 2;
    }

    for (long long i = 3; i <= sqrt(n); i += 2) {
        while (n % i == 0) {
            factors.push_back(i);
            n /= i;
        }
    }

    if (n > 2) {
        factors.push_back(n);
    }

    return factors;

}


}  // namespace prime_factors

File: src/dataset/python/prime-factors_exercism-ThomasZumsteg/gemini/GeminiFlash_prime_factors_v4.py
"""Finds the prime factors of a number"""

def prime_factors(number):
    """Finds prime factors of a number"""
    factors = []
    d = 2
    while d * d <= number:
        while number % d == 0:
            factors.append(d)
            number //= d
        d += 1
    if number > 1:
        factors.append(number)
    return factors